pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -261,5 +266,138 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
+
+	// For each app ingress
+	// For each item in the ps.applicationIngress
+	// Check if the appingress definition is the same as the item in the list
+	// If none of them match, do
+
+	for _, publishingStrategyIngress := range instance.Spec.ApplicationIngress {
+		for _, ingressController := range ingressControllerList.Items {
+			if doesIngressMatch(&publishingStrategyIngress, &ingressController) {
+				log.Info("match, move on to next cycle")
+				continue
+			} else {
+				// create a LocalObjectReference and fill it out secret passed in by OCM
+				newCertificate := &corev1.LocalObjectReference{
+					Name: publishingStrategyIngress.Certificate.Name,
+				}
+				if ingressController.Name == "default" {
+					// delete existing default ingresscontroller
+					err := r.client.Delete(context.TODO(), &ingressController)
+					if err != nil {
+						log.Error(err, "failed to delete existing ingresscontroller")
+						return reconcile.Result{}, err
+					}
+
+					// create a new ingresscontroller
+					newIngressControllerCR, err := newApplicationIngressControllerCR("default", string(publishingStrategyIngress.Listening), publishingStrategyIngress.DNSName, newCertificate)
+					if err != nil {
+						log.Error(err, "failed to generate data for new ingresscontroller CR")
+					}
+					err = r.client.Create(context.TODO(), newIngressControllerCR)
+					if err != nil {
+						log.Error(err, "failed to create new ingresscontroller CR")
+						return reconcile.Result{}, err
+					}
+					log.Info("succesfully created new ingresscontroller default")
+				}
+				// if the ingresscontroller name is not default
+				if ingressController.Name != "default" {
+					// if the content of this ingresscontroller matches
+					if doesIngressMatch(&publishingStrategyIngress, &ingressController) {
+						log.Info(fmt.Sprintf("%s already exists on cluster", ingressController.Name))
+						continue
+					} else { // if the ingresscontroller name is not default and the contents do not match
+						// create a new ingresscontroller
+						publishingStrategyIngressDNSName := publishingStrategyIngress.DNSName
+						firstPeriodIndex := strings.Index(publishingStrategyIngressDNSName, ".")
+						newIngressControllerName := publishingStrategyIngressDNSName[:firstPeriodIndex]
+						newIngressController, err := newApplicationIngressControllerCR(newIngressControllerName, string(publishingStrategyIngress.Listening), publishingStrategyIngress.DNSName, newCertificate)
+						if err != nil {
+							log.Error(err, "failed to generate data for new ingresscontroller CR")
+						}
+						err = r.client.Create(context.TODO(), newIngressController)
+						if err != nil {
+							log.Error(err, "failed to create new ingresscontroller CR")
+							return reconcile.Result{}, err
+						}
+						log.Info(fmt.Sprintf("succesfully create new ingresscontroller %s", newIngressControllerName))
+					}
+				}
+			}
```

Comment:
You should refactor this into at least two other function calls to make the logic easier to understand. 

Also when you start to use Else in go look back at your logic there is probably a simpler way to do it.